docs/testing/runtimes.md

```diff
@@ -0,0 +1,117 @@
+
+## Types of runtimes
+
+There are two types of runtimes used by Office Add-ins:
+
+- **JavaScript-only runtime**: A JavaScript engine supplemented with support for [WebSockets](https://developer.mozilla.org/docs/Web/API/WebSockets_API), [CORS (Cross-Origin Resource Sharing)](https://developer.mozilla.org/docs/Web/HTTP/CORS), and [local storage](https://developer.mozilla.org/docs/Web/API/Window/localStorage). 
+- **Browser runtime**: Includes all the features of a JavaScript-only runtime and adds a [rendering engine](https://developer.mozilla.org/docs/Glossary/Rendering_engine) that renders HTML.
+
+Details about these types are later in this article at [JavaScript-only runtime](#javascript-only-runtime) and [Browser runtime](#browser-runtime).
+
+The following table shows which type of runtime is used for the various possible features of an add-in. Note the choice of which type of runtime to use is an implementation detail that Microsoft could change at any time. The Office JavaScript Library doesn't assume that the same type of runtime will always be used for a given feature and your add-in architecture shouldn't assume this either.
+
+| Type of runtime | Add-in feature |
+|:-----|:-----|
+| JavaScript-only | Excel [custom function](../excel/custom-functions-overview.md)</br>(except when the runtime is [shared](#shared-runtime))</br></br>[Outlook event-based (autolaunched) task](../outlook/autolaunch.md)|
+| browser | [task pane](../design/task-pane-add-ins.md)</br></br>[dialog](../develop/dialog-api-in-office-add-ins.md)</br></br>[function command](../design/add-in-commands.md#types-of-add-in-commands)</br></br>Excel [custom function](../excel/custom-functions-overview.md)</br>(when the runtime is [shared](#shared-runtime))|
+
+A dialog always runs in its own process. So does an Outlook event-based task, also called autolaunched task. By default, task panes, function commands, and Excel custom functions each run in their own runtime process. However, for some Office host applications, the add-in manifest can be configured so that any two, or all three, of these run in the same runtime. See [Shared runtime](#shared-runtime).
+
+Depending on the host Office application and the features used in the add-in, there may be as many as six runtimes in an add-in, each running in its own process (but not necessarily running simultaneously). The following are examples.
+
+- A PowerPoint or Word add-in that that doesn't share any runtimes, and includes the following features, has three runtimes.
+
+  - A task pane
+  - A function command
+  - A dialog (A dialog can be launched from either the task pane or the function command. It is not a good practice to have multiple dialogs open simultaneously, but if you had one opened from the task pane and the function command at the same time, this add-in would have four runtimes.)
+
+- An Excel add-in that doesn't share any runtimes, and includes the following features, has four runtimes.
```

Comment:
I 'd say "... may have up to ...". We won't pre-create any runtime until it's needed.